partner_analytic/models/account_move.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2023 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+INV_TYPE_MAP = {
+    "out_invoice": "income",
+    "out_refund": "income",
+    "in_invoice": "expense",
+    "in_refund": "expense",
+}
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.onchange("product_id")
+    def _onchange_product_id(self):
+        res = super()._onchange_product_id()
+        inv_type = self.move_id.type
+        partner = self.move_id.partner_id
+        if (
+            partner
+            and inv_type
+            and inv_type in INV_TYPE_MAP
+            and not self.analytic_account_id
+        ):
+            an_accs = partner.get_partner_analytic_accounts()
+            an_acc = an_accs[INV_TYPE_MAP[inv_type]]
+            self.analytic_account_id = an_acc
+        return res
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if "move_id" in vals:
```

Comment:
To make it more readable you could first do if "move_id" not in vals: continue, and then do the rest

Reply:
Thank you 👍🏽